### ISSUE TITLE
Should we call gluon's panicking something else to distinguish it from rust's panicking?

### ISSUE THREAD

**Issue by Marwes** [Marwes/gluon#158]
I originally thought it was a good idea to call gluon's stack-unwinding errors `panic` to make it easily relatable to Rust. However when receiving a bug report over gitter it was initially confusing since it was unclear whether the panic was from rust or gluon (I suspected the former but wasn't 100%). The only other name I can think of is at the moment is `exception` but the way it is used currently it does not really match classical exceptions from say Java or C++ which may be confusing as well.

**Comment by DemiMarie** [comment]
Are they similar to Haskell exceptions?

**Comment by Marwes** [comment]
Yes, fairly similar to my knowledge. They can only be caught in the IO monad using the `catch` function.

**Comment by saifikhan** [comment]
What would you be modelling, Resumption Semantics or Termination Semantics ?

**Comment by Marwes** [comment]
@saifikhan As with a lot of things it is similar to Rust's semantics. It is possible to recover from "panics" and continue executing but it should only be used to prevent bugs from bringing down the program and not for normal control flow.

Does that answer your question?
